src/cpp/subscriber/StatisticsParticipantListener.cpp

```diff
@@ -341,6 +346,33 @@ void StatisticsParticipantListener::on_participant_discovery(
             }
         }
 
+        if (details::StatisticsBackendData::DiscoveryStatus::UPDATE != entity_discovery_info.discovery_status)
+        {
+            // Need to activate the meta traffic endpoint too
+            // Check if is the metatraffic endpoint already exists
+            std::shared_ptr<database::DataWriter> metatraffic_endpoint;
+            for (const auto& endpoint : participant->data_writers)
+            {
+                if (endpoint.second->name == metatraffic_prefix + "ENDPOINT_" + to_string(info.info.m_guid))
+                {
+                    metatraffic_endpoint = endpoint.second;
+                }
+            }
+            if (nullptr == metatraffic_endpoint)
+            {
+                logError(STATISTICS_BACKEND, "Participant " << to_string(
+                            info.info.m_guid) + " without meta-traffic endpoint")
+            }
+            else
```

Comment:
Could we keep a reference to the builtin endpoint inside the participant?
This way we could just get the reference here.